shared/src/types/address.rs

```diff
@@ -131,6 +137,73 @@ impl Address {
         }
     }
 
+    /// Convert an address to a fixed length utf-8 string bytes
+    fn to_fixed_len_string(&self) -> Vec<u8> {
+        match self {
+            Address::Established(EstablishedAddress { hash }) => {
+                format!("{}::{}", PREFIX_ESTABLISHED, hash)
+            }
+            Address::Implicit(ImplicitAddress::Ed25519(pkh)) => {
+                format!("{}::{}", PREFIX_IMPLICIT, pkh)
+            }
+            Address::Internal(internal) => match internal {
+                InternalAddress::PoS => internal::POS,
+                InternalAddress::PosSlashPool => internal::POS_SLASH_POOL,
+                InternalAddress::Ibc => internal::IBC,
+                InternalAddress::Parameters => internal::PARAMETERS,
+            }
+            .to_string(),
+        }
+        .into_bytes()
```

Comment:
an assertion that it's FIXED_LEN_STRING bytes might be nice

Reply:
Good idea, I've added debug assert here. There' a test for this `test_encoded_address_length` that checks all the addresses.